internal/resolution/entity_source_connector.go

```diff
@@ -0,0 +1,12 @@
+package resolution
+
+import (
+	"context"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+)
+
+type EntitySourceConnector interface {
+	ListEntities(ctx context.Context, catsrc *v1alpha1.CatalogSource) ([]*input.Entity, error)
```

Comment:
I think I'm missing something here. This still seems specific to OLMv0 catalog sources. Based on the PR description, it seemed like the goal was to remove the assumptions about the underlying API type.